easy_my_coop/models/account_invoice.py

```diff
@@ -100,18 +100,18 @@ def set_membership(self):
     def send_certificate_email(self, certificate_email_template, sub_reg_line):
         # we send the email with the certificate in attachment
         certificate_email_template.sudo().send_mail(self.partner_id.id, False)
-        
+
     def set_cooperator_effective(self, effective_date):
         sub_register_obj = self.env['subscription.register']
         share_line_obj = self.env['share.line']
 
+        certificate_email_template = self.get_mail_template_certificate()
+
         self.set_membership()
 
         sequence_operation = self.get_sequence_operation()
         sub_reg_operation = sequence_operation.next_by_id()
 
-        certificate_email_template = self.get_mail_template_certificate()
-

```

Comment:
I don't get what bring the change. You juste put the line some line upper. But from my understanding it doesn't make any difference.

Do I miss something ?

Reply:
Yep, the behavior is different since at line `self.set_membership()` we loose the information that we need to choose the email template.

Reply:
ok then ! I'll merge it